Begin addition of Alevin-fry processing information for single-nuclei and single-cell samples

allyhawkins opened the pull request:
This PR adds some initial processing information for alevin-fry and is stacked on #2. I began with one markdown file `processing_information.md `and filled out the headers to outline the information that I was thinking should be included in that document. I then used `make html` to build the `_build/html/index.html` file that can be viewed and contains the generated links to view the markdowns. The only files I directly edited myself were `processing_information.md` and `index.rst`.

For the actual documentation, I broke it up into two main sections, processing of single-cell and single-nuclei samples and processing of bulk RNA samples. Within the single-cell and single-nuclei samples I am thinking we can include sub sections about if a sample has an associated CITE-seq or spatial library.

For the section covering single-cell and single-nuclei samples, I chose to include an overview of the alevin-fry pipeline we are using followed by three sections covering index choice, alignment choice, and then count collapsing for single cell vs. single nuclei.

Here I am hoping for some general overview comments to help us get started building the rest of the documentation, in particular:

1. Does this organization make sense or do reviewers have other ideas for ways to organize these topics? Should we maybe split up this file more?

2. I also chose to include a few short sentences of why we chose Alevin-fry over cellranger. I wasn't sure at what level of detail we should be covering this topic, if at all?

3. I did not include any information about actually running the nextflow pipeline yet. I am thinking that will be a different section, do others agree or would we rather integrate that information into this text?

jaclyn-taroni reviewed (commented):
Thanks for getting this started @allyhawkins! I have some initial comments about organization to start:

- Let’s remove the intro paragraph for now. Much like a paper, we don’t really know what the “meat” of the text looks like just yet.
- Let’s reorganize the section that’s on scRNA-seq and snRNA-seq quantification, such that there is a paragraph that summarizes in a few sentences what people _really need to know_ and the other sections are more detailed for those who are inclined to dig in.

I think I will have quibbles about some of the wording and level of detail after those initial changes were made, but we should approach this iteratively!

allyhawkins commented:
@jaclyn-taroni Thanks for the feedback! I went ahead and removed the intro paragraph and moved the section that talked about the alevin-fry options like `--unfiltered-pl` to it's own section after the discussion of the index and alignment like you suggested. 

I also did some restructuring of the `splici` discussion including removing the section on actually generating the index. I tried to keep the discussion on the simpler side only including the points that I thought were relevant to understand what was actually done, but let me know if that is not the correct level of detail and if there are more or less concepts that should be included? 
I also think there may be points here where we would want to link to code or notebooks with more detailed information, like generating the splici index (but that would be for another PR).

jashapiro commented on docs/processing_information.md:
```diff
@@ -0,0 +1,64 @@
+# Processing Information
+
+## Single Cell and Single Nuclei Samples
```
This header seemed ambiguous for the info included. I also added hyphens because that seemed to be standard elsewhere, and changed to sentence case for the header, which I prefer, but we should make a decision about.
```suggestion
## Single-cell and single-nuclei RNA-seq
```

jashapiro commented on docs/processing_information.md:
```diff
@@ -0,0 +1,64 @@
+
+#### Filtering Cells
+
+### Samples with CITE-seq
```
Similar to a prior suggestion, making this less about samples and more about the processing. I anticipate the first sentence will be something like "For samples with CITE-seq data, we..."
```suggestion
### CITE-seq quantification
```

jashapiro reviewed (commented):
This looks like a good start! I left a number of comments throughout, so I'll try not to repeat myself too much. A general theme of my comments is to continue to push the "what we did" more than trying to explain exactly what the various options really are doing. I assume people will follow the links to the reference docs if they want to know more, or we would have a separate FAQ with some of the "why" answers. In general, I don't think those explanations make too much sense out of the context of what all the other options are, which would be way too much here!

The other theme I felt was about standardizing a few things:
- "single cell" vs. "single-cell"
- Title case vs. sentence case for headers
- capitalization and formatting of software names (italics, code format?)

allyhawkins commented:
Thanks for the feedback, @jashapiro. Based on your comments, I went ahead and made the decision to change everything to be in sentence case so it should be consistent. Also for use of `alevin-fry` and `salmon` in the text, I chose to use code formatting. If others feel strongly against this and would prefer a different format, I'm happy to change it. 

I also went through and cleaned up some of the areas that you had pointed out like the description of the splici index and the resolution description within the Alevin-fry parameters section.

jaclyn-taroni commented on docs/processing_information.md:
```diff
@@ -0,0 +1,60 @@
+
+#### Selective alignment
+
+We aligned reads to the transcriptome index using `salmon` with the defualt "selective alignment" strategy. 
```
:chants: spell check! spell check!

```suggestion
We aligned reads to the transcriptome index using `salmon` with the default "selective alignment" strategy. 
```

jaclyn-taroni commented on docs/processing_information.md:
```diff
@@ -0,0 +1,60 @@
+Briefly, selective alignment uses a mapping score validated approach to identify maximal exact matches between reads and the provided index. 
+For all samples, we used selective alignment to the `splici` index. 
+
+A more detailed description of the alignment strategy invoked by `salmon` in conjuction with `alevin-fry` can be found [here](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-02151-8). 
```
I would probably use an in text citation over `[here](<link to paper>)`.  I'll just get this out of the way and tell you that we're going to use the Genetics style citations, but I think you knew that based on the ones further up! DOI link for this one, too, please.

jaclyn-taroni commented on docs/processing_information.md:
```diff
@@ -0,0 +1,60 @@
+For our reference barcode list, we used a list of all possible cell barcodes from 10X Genomics.
+
+2. We chose to use the `cr-like-em` resolution strategy for [feature quantification and UMI de-duplication](https://alevin-fry.readthedocs.io/en/latest/quant.html). 
+Similar to the way Cell Ranger performs feature quantification, the `cr-like-em` resolution strategy assigns all UMI's that align to a single gene to that gene. 
```
I don't think this should be possessive

```suggestion
Similar to the way Cell Ranger performs feature quantification, the `cr-like-em` resolution strategy assigns all UMIs that align to a single gene to that gene. 
```

jaclyn-taroni commented on docs/processing_information.md:
```diff
@@ -0,0 +1,60 @@
+For all samples, we aligned FASTQ files to a reference transcriptome index referred to as the `splici` index.
+The [`splici` index](https://combine-lab.github.io/alevin-fry-tutorials/2021/improving-txome-specificity/) is built using transcripts from both spliced cDNA and intronic regions.
+Inclusion of intronic regions in the index used for alignment allowed us to capture both reads from mature, spliced cDNA and nascent, unspliced cDNA. 
+Alignment of RNA-sequencing data to an index containing intronic regions has been shown to reduce spuriously detected genes ([He _et al._ 2021](https://www.biorxiv.org/content/10.1101/2021.06.29.450377v1), [Kaminow _et al._ 2021](https://www.biorxiv.org/content/10.1101/2021.05.05.442755v1.full#sec-5))
```
I would probably use the DOI link for the He et al. citation if we don't think we need to tie it to v1. Since you're linking to a particular section for Kaminow et al., I think this is fine.

jaclyn-taroni commented on docs/processing_information.md:
```diff
@@ -0,0 +1,60 @@
+### Alignment and quantification using alevin-fry
+
+We used [`salmon alevin`](https://salmon.readthedocs.io/en/latest/alevin.html) and [`alevin-fry`](https://alevin-fry.readthedocs.io/en/latest/) to generate gene by cell counts matrices for all single-cell and single-nuclei samples.
+In brief, we utilized [selective alignment](#selective-alignment) to the [`splici` index](#the-splici-index) for all single-cell and single-nuclei samples. 
```
```suggestion
In brief, we utilized [selective alignment](#selective-alignment) to the [`splici` index](#reference-transcriptome-index) for all single-cell and single-nuclei samples. 
```

jaclyn-taroni reviewed (approved):
Much better! I had a few comments, but don't feel that I need another look. I reviewed Josh's earlier comments and it seems to me that they have all been addressed, so I think we can go ahead and merge this while he is away. I expect that there will be quite a few rounds of revisions for these docs overall!
